code-challenges.js

```diff
@@ -9,67 +9,185 @@
 var stringWithVowels2 = "ILoveJavaScript"
 
+const noVowels = (string) => {
+  let stringArray = string.split("")
+  let noVowelArray = stringArray.filter(value =>{
+    if (!['A','a','E','e','I','i','O','o','U','u'].includes(value)){
+      return true
+    }
+})
+  return noVowelArray.join("")
+}
+console.log(noVowels(stringWithVowels1));
+console.log(noVowels(stringWithVowels2));
 
 
+// // --------------------5) Copy/paste your code from #4. Refactor your code to include non-string edge cases. Inform your user that the variable passed into the vowel removal function is not a string.
+// // Use the test variables provided below. Expected output: "true is not a string" "42 is not a string" "mCdngMstr"
 
 
-// --------------------5) Copy/paste your code from #4. Refactor your code to include non-string edge cases. Inform your user that the variable passed into the vowel removal function is not a string.
-// Use the test variables provided below. Expected output: "true is not a string" "42 is not a string" "mCdngMstr"
-
 var refactorTester1 = true
 var refactorTester2 = 42
 var refactorTester3 = "IAmACodingMaster"
-
-
-
-
-
-// --------------------6) Create a function that takes in an array of strings and returns the string with the most characters.
-// Expected output: "chopsticks" "stemware"
-
+//
+
+//ADD before running string through split if else statement
+//check in if else statement for string, if not string -return "not a string "
+
+const noVowelString = (string) => {
+  if (typeof string !== 'string'){
+    return `${string} is not a string.`
+  }
+  let stringArray = string.split("")
+  let noVowelArray = stringArray.filter(value =>{
+    if (!['A','a','E','e','I','i','O','o','U','u'].includes(value)){
+      return true
+    }
+  })
+  return noVowelArray.join("")
+}
+
+console.log(noVowelString(refactorTester1));
+console.log(noVowelString(refactorTester2));
+console.log(noVowelString(refactorTester3));
+//
+//
+//
+//
+// // --------------------6) Create a function that takes in an array of strings and returns the string with the most characters.
+// // Expected output: "chopsticks" "stemware"
+//
 var utensils = ["fork", "knife", "tongs", "chopsticks", "skewer"]
 var vessels = ["cup", "mug", "stemware", "glass", "tumbler", "tea cup"]
+//
+//Make custom function that takes an array of strings as an arguement
+//Use sort to arrange arrays by length
+//return only the first word by using shift function on sorted variable.
+//log result
+
+//I originally was trying the pop function in a variety of patterns, with if/else statements but was coming up with errors or the wrong results.  I later came accross the sort function and thought that if I simply sort it by length and THEN execute the command shift to remove the first element and return just that it coul work!
+
+const mostChar = (array) => {
+    let charFilter = array.sort (function(a, b) {
+      return b.length - a.length;
+    })
+    return charFilter.shift ()
```

Comment:
awesome use of `shift()`  
also generally no space between the function name and executing parentheses 
```suggestion
    return charFilter.shift()
```